cluster/juju/prereqs/ubuntu-juju.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
```

Comment:
Always use:

```shell
  set -o errexit
  set -o nounset
  set -o pipefail
```

To ensure any errors will not go unnoticed.

You may need to work around some places where you can actually expect failure, e.g. use `|| :` after commands you expect might return non-zero and you're OK with it, and use `${varname-}` in cases where the `varname` variable might not be defined and you're OK with a blank instead.

The point is that it's explicit where you're OK with failures, the default makes it so that failures allow the script to keep running with potential disastrous effects.